packages/picasso.js/src/core/chart-components/ref-line/lines-and-labels.js

```diff
@@ -19,6 +19,67 @@ export function refLabelDefaultSettings() {
   };
 }
 
+function isMaxY(chart, slope, value) {
+  const scaleX = chart.scale('x');
+  const scaleY = chart.scale('y');
+  if (slope > 0) {
+    const maxY = scaleX.max() * slope + value;
+    if (maxY >= scaleY.max()) {
+      return true;
+    }
+  } else if (slope < 0) {
+    const minY = scaleX.max() * slope + value;
+    if (minY <= scaleY.min()) {
+      return true;
+    }
+  }
+  return false;
+}
+function getMaxXPosition(chart, slope, value) {
+  const scaleX = chart.scale('x');
+  const scaleY = chart.scale('y');
+  if (slope > 0) {
+    return scaleX((scaleY.max() - value) / slope);
+  } else if (slope < 0) {
+    return scaleX((scaleY.min() - value) / slope);
+  }
+}
+
+function getFormatter(p) {
+  let formatter;
+  if (typeof p.formatter === 'string') {
+    formatter = chart.formatter(p.formatter);
```

Comment:
Is`chart` a global variable?

Reply:
Also what is the difference between this line and line 53?

Reply:
chart needs to be passed to the function, that I shall fix but the other part I have not touched / changed as it was already existing code, I have just moved it to a function so I can call it in two places.